protocols/raft/src/test/java/io/atomix/protocols/raft/storage/snapshot/impl/FileSnapshotStoreTest.java

```diff
@@ -189,4 +192,43 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc)
     }
     testId = UUID.randomUUID().toString();
   }
+
+  /** Tests writing a snapshot. */
+  @Test
+  public void testWriteSnapshotChunks() {
+    final SnapshotStore store = createSnapshotStore();
+    final WallClockTimestamp timestamp = new WallClockTimestamp();
+    final Snapshot snapshot = store.newSnapshot(2, 1, timestamp);
+    assertEquals(2, snapshot.index());
+    assertEquals(timestamp, snapshot.timestamp());
+
+    assertNull(store.getSnapshot(2));
+
+    try (SnapshotWriter writer = snapshot.openWriter()) {
+      writer.writeLong(10);
+    }
+
+    assertNull(store.getSnapshot(2));
+
+    try (SnapshotWriter writer = snapshot.openWriter()) {
+      writer.writeLong(11);
+    }
+
+    assertNull(store.getSnapshot(2));
+
+    try (SnapshotWriter writer = snapshot.openWriter()) {
+      writer.writeLong(12);
+    }
+
+    assertNull(store.getSnapshot(2));
```

Comment:
Why you assert this every time? :sweat_smile:

Reply:
Might make sense to split the test to test the different concerns like - 

 1. snapshot is not completed when I close writer
 2. snapshot is not completed/written after I wrote into it
 3. when snapshot complete it exists on disk etc.

Reply:
This isn't really new code, I just merged two test files, I didn't write those tests :sweat_smile:

Reply:
Ok but maybe we can improve here :sweat_smile:

Reply:
:+1:

Reply:
I'll add new tests but I won't rewrite the other ones tbh, as this isn't an implementation we plan to use in the long run. So the asserts remain :see_no_evil: